src/main/java/practice/CandidateValidator.java

```diff
@@ -1,5 +1,20 @@
 package practice;
 
-public class CandidateValidator {
-    //write your code here
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+    private static final int MINIMUM_AGE = 35;
+    private static final int MINIMUM_AGES_IN_UKRAINE = 10;
+    private static final String UKRAINIAN = "Ukrainian";
+
+    @Override
+    public boolean test(Candidate candidate) {
+        int toYearLivingInUkraine = Integer.parseInt(candidate.getPeriodsInUkr().substring(5));
+        int fromYearLivingInUkraine = Integer.parseInt(candidate.getPeriodsInUkr().substring(0,4));
+        int yearsLivingInUkraine = toYearLivingInUkraine - fromYearLivingInUkraine;
```

Comment:
```suggestion
        String[] periodInUkraine = candidate.getPeriodsInUkr().split("-");
        int yearsLivingInUkraine = Integer.parseInt(periodInUkraine[1])
                - Integer.parseInt(periodInUkraine[0]);
```
Also create constants fro array indexes